src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -2352,7 +2353,7 @@ class LoadDataSegment(BaseSegment):
         "INTO",
         "TABLE",
         Ref("TableReferenceSegment"),
-        Sequence("PARTITION", Ref("SelectPartitionClauseSegment"), optional=True),
+        Ref("SelectPartitionClauseSegment", optional=True),
```

Comment:
Can you explain this one? Docs seem to suggest it has the word PARTITION: https://dev.mysql.com/doc/refman/8.0/en/load-data.html

Reply:
SelectPartitionClauseSegment is introduced in e85fcbb1be134a46fe3444fc850caa0b3287d5ee (CommitDate: Jul 2 2021) and it does include the word "PARTITION" in inside.
LoadDataSegment is introduced in e81993fa9a5cb82636b1b2055e9ff42912707bd5 (CommitDate: Jun 29 2022) and misuse SelectPartitionClauseSegment (The misuse was overlooked because test was not included).
I noticed that in the process of implementing ReplaceSegment, fixed it, and add a test to load_data.sql.

Reply:
Cool, it makes sense now, thank you!